clang/test/SemaSYCL/deferred-diagnostics-emit.cpp

```diff
@@ -2,13 +2,18 @@
 //
 // Ensure that the SYCL diagnostics that are typically deferred are correctly emitted.
 
+namespace std {
```

Comment:
Do you validate variable templates anywhere?  How about alias templates?

Reply:
I put those over in sycl-restrict.cpp, along with checks for auto, typedef, and some false postives.  Let me know if you have any cases to add.

In this file, we're just exercising that the deferred diagnostics are working when the kernel lambda is itself templated.

Reply:
I don't see any variable template or alias template examples over there.  Can you point them out please?

Reply:
I misunderstood you the first time. I have added alias templates and  C++14 variable templates to sycl-restrict.cpp (starting at line 116) as both cases that should be detected and possible false positive cases that should not be flagged.  

If you see anything else that we should check, let me know.